examples/ota-provider-app/linux/main.cpp

```diff
@@ -238,6 +280,31 @@ int main(int argc, char * argv[])
     }
 
     chip::DeviceLayer::ConfigurationMgr().LogDeviceConfig();
+
+    if (gSetupDiscriminator.HasValue())
+    {
+        // Set discriminator to user specified value
+        ChipLogProgress(SoftwareUpdate, "Setting discriminator to: %" PRIu16, gSetupDiscriminator.Value());
+        err = chip::DeviceLayer::ConfigurationMgr().StoreSetupDiscriminator(gSetupDiscriminator.Value());
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(SoftwareUpdate, "Setup discriminator setting failed with code: %" CHIP_ERROR_FORMAT, err.Format());
+            return 1;
+        }
+    }
+
+    if (gSetupPasscode.HasValue())
+    {
+        // Set passcode to user specified value
+        ChipLogProgress(SoftwareUpdate, "Setting passcode to: %" PRIu32, gSetupPasscode.Value());
+        err = chip::DeviceLayer::ConfigurationMgr().StoreSetupPinCode(gSetupPasscode.Value());
```

Comment:
Does this actually work? I know for `StoreSetupDiscriminator`, @selissia had to doctor up something for ReadConfigValue in PosixConfig for Darwin